find_me_a_sandwich/app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,27 @@
+class FavoritesController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+  end
+
+  def friends
+    @friends = User.visible.where.not(id: current_user.id)
+    @friends_favorites, @total_count = Favorite.friends_favorites(@friends)
+  end
+
+  def create
+    Favorite.create(user_id: current_user.id,
+                    menu_item_id: favorite_params[:menu_item_id])
+    redirect_to :back
+  end
+
+  def destroy
+    Favorite.where(user_id: current_user.id,
+                   menu_item_id: favorite_params[:menu_item_id]).delete_all
```

Comment:
There should only be one favorite per user/menu item combo.  Use `#find_by` instead of `#where` to ensure only one record is returned.